lib/screens/authenticate/sign_in.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:helping_hands/services/auth.dart';
+import 'package:helping_hands/shared/constants.dart';
+import 'package:helping_hands/shared/loading.dart';
+import 'package:flutter/material.dart';
+
+class SignIn extends StatefulWidget {
+
+  final Function toggleView;
+  SignIn({ this.toggleView });
+
+  @override
+  _SignInState createState() => _SignInState();
+}
+
+class _SignInState extends State<SignIn> {
+
+  final AuthService _auth = AuthService();
+  final _formKey = GlobalKey<FormState>();
+  String error = '';
+  bool loading = false;
+
+  // text field state
+  String email = '';
+  String password = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return loading ? Loading() : Scaffold(
+      backgroundColor: Colors.brown[100],
+      appBar: AppBar(
+        backgroundColor: Colors.green[600],
+        elevation: 0.0,
+        title: Text('Sign in to Helping Hands'),
+        actions: <Widget>[
+          FlatButton.icon(
+            icon: Icon(Icons.person),
+            label: Text('Register'),
+            onPressed: () => widget.toggleView(),
+          ),
+        ],
+      ),
+      body: Container(
+        padding: EdgeInsets.symmetric(vertical: 20.0, horizontal: 50.0),
+        child: Form(
+          key: _formKey,
+          child: Column(
+            children: <Widget>[
+              SizedBox(height: 20.0),
+              TextFormField(
+                decoration: textInputDecoration.copyWith(hintText: 'email'),
+                validator: (val) => val.isEmpty ? 'Enter an email' : null,
+                onChanged: (val) {
+                  setState(() => email = val);
+                },
+              ),
+              SizedBox(height: 20.0),
+              TextFormField(
+                obscureText: true,
+                decoration: textInputDecoration.copyWith(hintText: 'password'),
+                validator: (val) => val.length < 8 ? 'Enter a password 8+ chars long' : null,
+                onChanged: (val) {
+                  setState(() => password = val);
+                },
+              ),
+              SizedBox(height: 20.0),
+              RaisedButton(
+                color: Colors.green[600],
+                child: Text(
+                  'Sign In',
+                  style: TextStyle(color: Colors.white),
+                ),
+                onPressed: () async {
```

Comment:
We should split all functions out of the return call. This can be defined before the build function and called from here. By splitting the functions out we can write automated tests to test that they work.

Reply:
This is still not addressed.